### ISSUE TITLE
Proposal for Partial synchronization

### ISSUE THREAD

**Issue by sduchesneau** [dfuse-io/dfuse-eosio#48]
# Context

Historically, all dfuse components were created with the assumption that the whole blockchain would be processed. A lot of effort has been put to allow parallel reprocessing of all components (block files, database, etc.)

Now that users are running local instances of dfuse for EOSIO, it makes sense to allow them to sync a chain only partially, from a specific block (usually close to HEAD). Parts of it can already be done, but there are limitations.

# What works currently

With this pull request: https://github.com/dfuse-io/dfuse-eosio/pull/47, 

1. You can sync with kylin within 1h (depending on the age of your source state snapshot)
2. You can have a working 'search' in a given range (graphql) -- without decoded dbops --
3. You can have a working "live" search on the head (graphql) -- without decoded dbops --
4. You can get "block" or get "blockref by time range" (graphql) 

# What does  NOT work currently

1. Fluxdb currently requires syncing from the beginning of the chain, because it saves ALL historical changes. This would mean generating ALL block files (nodeos+mindreader running from block 1, or in parallel from different snapshots...)

2. eosws (serving all websocket/REST) has _not been tested_ with partial chain...

3. abicodec requires a working fluxdb, this prevents decoding the dbops on search results

# Proposal

* FluxDB could bootstrap from a state snapshot (decoding all objects directly from that file). It would require FluxDB to have the concept of a "low_block_num", at which it has that snapshot.
The hardest part is the decoding from the state snapshots, the behavior modifications in fluxdb should be trivial.

This would allow abicodec to work

(Eosws has some calls depending on fluxdb, more discovery work is needed.)

**Comment by sduchesneau** [comment]
note: pr #47 is now merged into develop
